s3same/iam.py

```diff
@@ -0,0 +1,75 @@
+from botocore.exceptions import ClientError
+
+IAMName = 's3same_travis'
+
+def _policy_string(bucket):
+    return """{
+      "Version": "2012-10-17",
+      "Statement": [
+        {
+          "Action": [
+            "s3:ListBucket"
+          ],
+          "Effect": "Allow",
+          "Resource": [
+            "arn:aws:s3:::%(bucket)s"
+          ]
+        },
+        {
+          "Action": [
+            "s3:PutObject",
+            "s3:PutObjectAcl"
+          ],
+          "Effect": "Allow",
+          "Resource": [
+            "arn:aws:s3:::%(bucket)s/*"
+          ]
+        }
+      ]
+    }""" % dict(bucket=bucket)
+
+def _policy_arn(iam, bucket):
+    policy_arn = None
+    try:
+        response = iam.create_policy(
+                PolicyName=IAMName,
+                PolicyDocument=_policy_string(bucket),
+                )
+        policy_arn = response.get('Policy', {}).get('Arn')
+    except ClientError as e:
+        if e.response['Error']['Code'] != 'EntityAlreadyExists':
+            raise e
+        marker = None
+        while not policy_arn:
+            response = iam.list_policies(Scope='Local')
+            policy_arn = next((p.get('Arn') for p in response.get('Policies', []) if p.get('PolicyName') == IAMName), None)
+            if not response.get('IsTruncated'):
+                break
+            marker = response['Marker']
```

Comment:
I think you're right that I can move it out of the `except` clause as-is, but I think if I use `while True:`, then it'd need a condition around it or to be inside the `except` clause...

The logic is this:
1. Try to create the policy.
2. If creation fails, but not because it already exists, re-raise.
3. If creation fails because it already exists, then we need to get its ARN, but there isn't a direct way to do that, so we have to get all the policies, which is a paged endpoint, and use the ARN of the first one with a matching name.

My understanding is that the generator version should be a little better than the list comprehension version, since it won't do any work until the `next()` is evaluated and it'll stop going through the list of policies as soon as it finds a match.